test/uk/gov/hmcts/pipeline/CVEPublisherTest.groovy

```diff
@@ -0,0 +1,49 @@
+package uk.gov.hmcts.pipeline
+
+import com.microsoft.azure.documentdb.DocumentClient
+import spock.lang.Specification
+import uk.gov.hmcts.contino.JenkinsStepMock
+
+class CVEPublisherTest extends Specification {
+
+  CVEPublisher cvePublisher
+  def steps
+
+  def documentClient
+
+  def setup() {
+    documentClient = Mock(DocumentClient.class)
+    steps = Mock(JenkinsStepMock.class)
+    steps.getEnv() >> [
+      BRANCH_NAME: 'master',
+      GIT_URL: 'http://example.com',
+      COSMOSDB_TOKEN_KEY: 'abcd'
+    ]
+    def closure
+    steps.withCredentials(_, { closure = it }) >> { closure.call() }
+
+    cvePublisher = new CVEPublisher(
+      steps,
+      false,
+      documentClient
+    )
+  }
+
+  def "Publishing CVE report does not throw unhandled error"() {
+    given:
+    def report = [
+      dependencies: [
+        something: '1.0'
+      ]
+    ]
+
+    when:
+    cvePublisher.publishCVEReport(report)
+
+    then:
+    notThrown(Exception.class)
+
+    1 * documentClient.createDocument('dbs/jenkins/colls/cve-reports', _, null, false)
```

Comment:
I tried for far too long to verify that the report is correct but it doesn't work,
I think it's because the `Document` class doesn't implement equals.

could extract something else out but seems overkill